src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -502,6 +503,7 @@ public static object ConvertNativeComInterfaceToManaged(IntPtr pUnk)
 #endif
         }
 
+        [RequiresDynamicCode("Marshalling code for the object might not be available. Use the SizeOf<T> overload instead.")]
```

Comment:
Could you suppress the occurrences in this file instead? These methods are called from compiler generated code only (`CompilerHelpers` namespace). The compiler is responsible to emit diagnostics for unfriendly interop.